### DISCUSSION TITLE
minimum viable phoneme coverage / utterance sample set for training a custom voice model

### DISCUSSION THREAD

**Discussion by thoraxe** [coqui-ai/TTS#872] [General Q&A]
I've seen a lot of "unscientific" (heavy air quotes) comments about "you need at least 100 samples" or "you need X hours" when it comes to custom voice training models. I have been digging around and have not really found any science that specifically studies the minimum viable coverage of phonemes.

I just put this into the documentation ideas:
https://github.com/coqui-ai/TTS/discussions/799#discussioncomment-1444388

The closest I've found so far looks like a repository of 1,100 or so English sentences, which is still a tremendous amount of content:

https://github.com/rhasspy/tts-prompts

The models are getting better and better, and faster and faster. There are only so many phonemes in the English language (44). It seems curious to me that there is not a "defacto" set of utterances that come together to make a minimum viable set for training a new voice model.

How is it that there are only 44 phonemes but we still need several hours of audio to train a model? I understand that there is a lot of nuance in the way that different phonemes are combined. It's still surprising that there's no meaningful starting point that's in the several hundred (or less) ranges.

Any thoughts here?

**Comment by domcross** [comment]
For the 44 phonemes there are 1936 bi-grams and 85184 tri-grams possible. Of course not all combinations appear in actual language, but nevertheless there will be more than a few hundred phrases necessary.

**Reply by thoraxe** [reply]
While what you're saying is true, it falls into the "you need several hours of dialog" trap of being non-specific ("there will be more than a few hundred phrases necessary").

Of the 44 phonemes, 1936 bi-grams, and 85184 tri-grams in English, how many are used in actual language, and what is the minimum set of utterances required to cover them?

It sounds more and more like this is not an area of the research that's been covered, or I don't know how to find it.

**Reply by domcross** [reply]
Exactly, this is a recommendation from best-practice efforts or - expressed in a more scientific way: heuristics derived from experiments. I suppose the dataset size (number of phrases required) is also depending on the model and method you are training (e.g. transfer learning)
